chat-iOS/Views/Chats/ChatsViewModel.swift

```diff
@@ -5,14 +5,62 @@
 //  Created by jun on 2020/06/22.
 //
 
+import Firebase
+
 protocol ChatsViewModelProtocol {
     var presenter: ChatsViewModelOutput! { get set }
+    var firestore: Firestore! { get set }
+    
+    func setUpFirestore()
+    func fetchTransScript()
 }
 
 protocol ChatsViewModelOutput {
-    
+    func successFetchTransScript(transScripts: [Transcript])
 }
 
 final class ChatsViewModel: ChatsViewModelProtocol {
+    var firestore: Firestore!
```

Comment:
firestoreの方は他クラスから参照しないと思うのでprivateにした方がいいです！

Reply:
firestoreはprivateでは宣言できないのでinternalにしました。
privateにした場合は以下のエラーが出ます
```
Property 'firestore' must be declared internal because it matches a requirement in internal protocol 'ChatsViewModelProtocol'
```

Reply:
そこに関してはプロトコルの仕様(概念)の話に戻るのですが、プロトコルというのは内部実装を外にできる限り見せずに抽象化して他のクラスに公開するために使うもの(他のクラスは自クラスの実装について知る必要がない)で、プライベートメソッドやプロパティは内部実装の話になってしまうのでプロトコルに入れるというのがそもそもおかしな話になってしまいます。なのでそのエラーに関してはプロトコルから削除すればコンパイルは通ると思います！！

要は外部からの参照が必要なプロパティー、メソッドのみプロトコルに入れるのがいいと思います！